src/dodal/devices/i04/murko_results.py

```diff
@@ -0,0 +1,221 @@
+        self._last_omega = 0
+        self._last_result = None
+        self.sample_id = soft_signal_rw(str)  # Should get from redis
+        self.coords = {"x": {}, "y": {}, "z": {}}
+        self.search_angles = OrderedDict(
+            [  # Angles to search and dimensions to gather at each angle
+                (90, ("x", "z")),
+                (180, ("x", "y")),
+                (270, ()),  # Stop searching here
+            ]
+        )
+        self.angles_to_search = list(self.search_angles.keys())
+
+        with self.add_children_as_readables():
+            # Diffs from current x/y/z
+            self.x_mm, self._x_mm_setter = soft_signal_r_and_setter(float)
+            self.y_mm, self._y_mm_setter = soft_signal_r_and_setter(float)
+            self.z_mm, self._z_mm_setter = soft_signal_r_and_setter(float)
+        super().__init__(name=name)
+
+    @AsyncStatus.wrap
+    async def stage(self):
+        # Subscribe to redis pub/
+        await self.pubsub.subscribe("murko-results")
+        self._x_mm_setter(None)
+        self._y_mm_setter(None)
+        self._z_mm_setter(None)
+
+    @AsyncStatus.wrap
+    async def unstage(self):
+        # Unsubscribe
+        await self.pubsub.unsubscribe()
+
+    def get_beam_centre(self) -> tuple[int, int]:
+        return (442, 363)
+
+    def get_coords_if_at_angle(
+        self,
+        metadata: MurkoMetadata,
+        result: MurkoResult,
+        omega: float,
+        search_angle: float,
+    ) -> np.ndarray | None:
+        LOGGER.info(f"Compare {omega}, {search_angle}, {self._last_omega}")
+        if (  # if last omega is closest
+            abs(omega - search_angle) >= abs(self._last_omega - search_angle)
+            and self._last_result is not None
+        ):
+            closest_result = self._last_result
+            closest_omega = self._last_omega
+        elif omega - search_angle >= 0:  # if this omega is closest
+            closest_result = result
+            closest_omega = omega
+        else:
+            return None
+        coords = closest_result[
+            "most_likely_click"
+        ]  # As proportion from top, left of image
+        shape = closest_result["original_shape"]  # Dimensions of image in pixels
+        # Murko returns coords as y, x
+        centre_px = (coords[1] * shape[1], coords[0] * shape[0])
+        LOGGER.info(
+            f"Using image taken at {closest_omega}, which found xtal at {centre_px}"
+        )
+
+        beam_dist_px = calculate_beam_distance(
+            self.get_beam_centre(),  # beam centre
+            centre_px[0],
+            centre_px[1],
+        )
+
+        return camera_coordinates_to_xyz_mm(
+            beam_dist_px[0],
+            beam_dist_px[1],
+            closest_omega,
+            metadata["microns_per_x_pixel"],
+            metadata["microns_per_y_pixel"],
+        )
+
+    @AsyncStatus.wrap
+    async def trigger(self):
+        # May be better as kickoff/complete?
```

Comment:
Should: Leaving comments like this just confuses future devs. We should either:
* Make it a `kickoff/complete` if we think it's better
* Decide that trigger is best and remove the comment
* If we want more discussion make an issue to discuss it

In all cases the comment should be removed